### ISSUE TITLE
REPEND crashes when using TRIM on the target series.

### ISSUE THREAD

**Issue by dockimbel** [red/red#4281]
The following code crashes the console:
```
data: "   123   "
repend data [trim data]
```
If `trim` is removed, it works as expected.

**Comment by dockimbel** [comment]
Most probably related to `reduce/into`.
See also https://github.com/red/red/issues/3340

**Comment by dockimbel** [comment]
Crash happens in `string/alter` in:
```
if mode = MODE_INSERT [
	move-memory				
		(as byte-ptr! s1/offset) + h1 + offset + size
		(as byte-ptr! s1/offset) + h1 + offset
		(as-integer s1/tail - s1/offset) - h1
]
```
The last argument to `move-memory` is negative (`-6` in above case), because `trim data` makes the `tail data` reference become "past-end".

The fix for this issue should be part of the general solution for handling "past-end" series internally.

**Comment by hiiamboris** [comment]
https://github.com/red/red/pull/4097

**Comment by dockimbel** [comment]
@hiiamboris I will process the PRs once the lexer branch is merged into master.

**Comment by dockimbel** [comment]
That code does not crash anymore (probably since a while). I'll add unit tests and close it.
